integration-tests/src/tests/client/sync_state_nodes.rs

```diff
@@ -45,8 +45,13 @@ fn sync_state_nodes() {
         let mut near1 = load_test_config("test1", port1, genesis.clone());
         near1.network_config.peer_store.boot_nodes = convert_boot_nodes(vec![]);
         near1.client_config.min_num_peers = 0;
+
+        let _dir1 = Arc::new(tempfile::Builder::new().prefix("sync_nodes_1").tempdir().unwrap());
```

Comment:
It would be helpful to leave a comment here explaining the observed issues with directories being cleaned up too soon.